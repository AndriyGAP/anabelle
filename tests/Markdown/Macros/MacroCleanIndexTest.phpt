<?php

declare(strict_types=1);

namespace Ublaboo\Anabelle\Tests\Console\Utils;

require_once __DIR__ . '/../../bootstrap.php';

use Tester\Assert;
use Tester\TestCase;
use Ublaboo\Anabelle\Markdown\Macros\MacroCleanIndex;

class MacroCleanIndexTest extends TestCase
{

	public function testHeading(): void
	{
		$content = "\n\n## Nope\n# Test heading\n\n# Foo\n\n";

		$macro = new MacroCleanIndex;
		$macro->runMacro('', '', $content);

		Assert::same("# Test heading\n\n", $content);
	}


	public function testSections(): void
	{
		$content = "# Test heading\n@ home:home.md\nNothing\n@ about:about.md\n@@ method1:method1.md\nFoo\n@@ method1:method1.md\n";

		$macro = new MacroCleanIndex;
		$macro->runMacro('', '', $content);

		Assert::same("# Test heading\n\n@ home:home.md\n@ about:about.md\n@@ method1:method1.md\n@@ method1:method1.md\n", $content);
	}
}

(new MacroCleanIndexTest())->run();
