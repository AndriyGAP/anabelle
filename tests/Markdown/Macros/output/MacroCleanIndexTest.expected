# Test heading
@ home:home.md

@ about:about.md
@@ method1:method1.md
@@ method1:method1.md
